### PR TITLE
Add verification to rate limiter number field

### PR DESCRIPTION
**Pull request by b4sjoo**
### Description
This PR will add verification to rate limiter number field to avoid illegal input

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: common/src/main/java/org/opensearch/ml/common/controller/MLRateLimiter.java [hunk 1]
```diff
@@ -49,6 +50,14 @@ public static MLRateLimiter parse(XContentParser parser) throws IOException {
             switch (fieldName) {
                 case LIMIT_FIELD:
                     limit = parser.text();
+                    try {
+                        int limitNumber = Integer.parseInt(limit);
```
Comment (jngz-es) [inline common/src/main/java/org/opensearch/ml/common/controller/MLRateLimiter.java:54]: What if the number is zero?

Comment (b4sjoo) [reply common/src/main/java/org/opensearch/ml/common/controller/MLRateLimiter.java:54]: Zero will simply build a rate limiter that refuse all requests, so essentially this is a legal case.

Comment (jngz-es) [reply common/src/main/java/org/opensearch/ml/common/controller/MLRateLimiter.java:54]: Makes sense. Do we support fraction, like 1.5?

Comment (b4sjoo) [reply common/src/main/java/org/opensearch/ml/common/controller/MLRateLimiter.java:54]: Nice catch, I just test, fraction works so I change int into double

**Review by jngz-es — APPROVED** [review]

**Review by ylwu-amzn — APPROVED** [review]

File: common/src/main/java/org/opensearch/ml/common/controller/MLRateLimiter.java [hunk 2]
```diff
@@ -49,6 +50,14 @@ public static MLRateLimiter parse(XContentParser parser) throws IOException {
             switch (fieldName) {
                 case LIMIT_FIELD:
                     limit = parser.text();
+                    try {
+                        double limitNumber = Double.parseDouble(limit);
```
Comment (samuel-oci) [inline common/src/main/java/org/opensearch/ml/common/controller/MLRateLimiter.java:54]: shouldn't we add validation for `UNIT_FIELD` as well?

**Pull request merged**